### ISSUE TITLE
Repaint artificats problem when drag window

### ISSUE THREAD

**Issue by vaniusrb** [linuxmint/cinnamon#8582]
-  Cinnamon 4.0.10
-  Mint 19.1 Tessa
-  Graphics hardware "Intel Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics" (Intel i7 4790) driver: "intel"
-  64 bit
-  Kernel: 5.0.10-050010-generic x86_64
- /home/<username>/.xsession-errors: [xsession-errors.log](https://github.com/linuxmint/cinnamon/files/3214180/xsession-errors.log)

**Issue**

When I move window frame I get a "repaint" problem.
In .xsession-errors file there are some CRITICAL and WARNING records.

![cinnamon-drag-window-artifact](https://user-images.githubusercontent.com/11839284/58287043-75fb3d80-7d86-11e9-8c76-ab58d44d2ce6.png)

**Steps to reproduce**

**Expected behaviour**

**Other information**

**Comment by scoson** [comment]
`System:
  Host: scott-Inspiron-3558 Kernel: 4.15.0-50-generic x86_64 bits: 64 
  compiler: gcc v: 7.3.0 Desktop: Cinnamon 4.0.10 
  Distro: Linux Mint 19.1 Tessa base: Ubuntu 18.04 bionic 
Machine:
  Type: Laptop System: Dell product: Inspiron 3558 v: N/A serial: <filter> 
  Mobo: Dell model: 0PNFDX v: A00 serial: <filter> UEFI: Dell v: A17 
  date: 08/27/2018 
Battery:
  ID-1: BAT0 charge: 41.4 Wh condition: 35.2/41.4 Wh (85%) 
  model: SMP DELL VN3N047E status: Full 
CPU:
  Topology: Dual Core model: Intel Core i3-5015U bits: 64 type: MT MCP 
  arch: Broadwell rev: 4 L2 cache: 3072 KiB 
  flags: lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 16761 
  Speed: 1497 MHz min/max: 500/2000 MHz Core speeds (MHz): 1: 1996 2: 1996 
  3: 1995 4: 1995 
Graphics:
  Device-1: Intel HD Graphics 5500 vendor: Dell driver: i915 v: kernel 
  bus ID: 00:02.0 
  Display: x11 server: X.Org 1.19.6 driver: modesetting unloaded: fbdev,vesa 
  resolution: 1366x768~60Hz 
  OpenGL: renderer: llvmpipe (LLVM 7.0 256 bits) v: 3.3 Mesa 18.2.8 
  direct render: Yes 
Audio:
  Device-1: Intel Broadwell-U Audio vendor: Dell driver: snd_hda_intel 
  v: kernel bus ID: 00:03.0 
  Device-2: Intel Wildcat Point-LP High Definition Audio vendor: Dell 
  driver: snd_hda_intel v: kernel bus ID: 00:1b.0 
  Sound Server: ALSA v: k4.15.0-50-generic 
Network:
  Device-1: Intel Wireless 3160 driver: iwlwifi v: kernel port: 4040 
  bus ID: 06:00.0 
  IF: wlp6s0 state: up mac: <filter> 
  Device-2: Realtek RTL810xE PCI Express Fast Ethernet vendor: Dell 
  driver: r8169 v: 2.3LK-NAPI port: 3000 bus ID: 07:00.0 
  IF: enp7s0 state: down mac: <filter> 
Drives:
  Local Storage: total: 931.51 GiB used: 591.73 GiB (63.5%) 
  ID-1: /dev/sda vendor: Toshiba model: MQ01ABD100 size: 931.51 GiB 
Partition:
  ID-1: / size: 137.75 GiB used: 16.94 GiB (12.3%) fs: ext4 dev: /dev/sda6 
Sensors:
  System Temperatures: cpu: 36.0 C mobo: 38.0 C 
  Fan Speeds (RPM): cpu: 0 
Info:
  Processes: 195 Uptime: 5h 03m Memory: 3.76 GiB used: 1.15 GiB (30.5%) 
  Init: systemd runlevel: 5 Compilers: gcc: 7.4.0 Shell: bash v: 4.4.19 
  inxi: 3.0.27`

I'm getting this the last few days too. It seems like since the new Mint updates along with the 4.15.0-50 kernel. I tried booting back to 4.15.0-48 but that didn't help. As shown here I also get this effect on main menu (using Void or Divo) as well as any window..

![Screenshot from 2019-05-23 18-28-18](https://user-images.githubusercontent.com/46092441/58292884-c378aa00-7d88-11e9-835c-4729b3bb2097.png)

**Comment by jaszhix** [comment]
> DESKTOP_SESSION=cinnamon2d
> Cjs-Message: 17:41:19.092: JS LOG: ACTIVATING SOFTWARE RENDERING

This will break the window manager if you're not booting with the `nomodeset` kernel flag, which I think is really the only supported case for software rendering. The fallback mode sets a few different flags for software rendering to work correctly. If Cinnamon starts in software rendering without environmental configuration, it means a wrong or misconfigured graphics driver is being used. Using the GPU will use less power, and will ensure you're using Cinnamon the way it was tested during development.

> MUFFIN_NO_SHADOWS=1

Try removing this from your environment and see if the issue improves.

**Comment by scoson** [comment]
@jaszhix thank you for replying. I had to do a quick search to find where to change this, but it appears my environment file doesn't have that line anyway, only a line concerning a few paths. But, full disclosure overnight I threw caution to the wind and installed the dailybuilds ppa to see if the newer Cinnamon build would fix it somehow. I didn't know there would be any help any time soon. I'm considering just doing a clean re-install today to see if that helps. I keep good backups so it's not that painful..

**Comment by jaszhix** [comment]
Unfortunately, I don't think this will improve in 4.2. Since we switched to a newer version of Cogl (Clutter's OpenGL library) in 4.0, software rendering has been flakey. This is probably confusing when already dealing with driver issues. I never have to deal with the Intel driver on skylake hardware, it just works for me, so my experience other than that is limited for Intel.

[People used to use software rendering as a workaround](https://github.com/linuxmint/cinnamon/issues/7985) in <=3.8, but we're better off fixing the performance issues that drove people to hack up their environment. :slightly_smiling_face:

**Comment by scoson** [comment]
Thank you for the info jaszhix, hopefully vaniusrb has better luck with this. For myself, should I just go ahead and try a clean re-install? Maybe I don't have enough horsepower for Linux Mint Cinnamon now?

**Comment by jaszhix** [comment]
It's more like your computer's resources are not being used. The driver would appear to not be working correctly. You could try enabling the the HWE stack.

`sudo apt install xserver-xorg-hwe-18.04 --install-recommends --dry-run`

This will install a newer LTS-supported kernel and xorg components provided by Ubuntu.

Remove any extra kernel flags from /etc/default/grub -> `GRUB_CMDLINE_LINUX_DEFAULT=`, clear any custom configuration in /etc/X11/xorg.conf.d or /etc/X11/xorg.conf, run `sudo update-grub`. Just where I would start poking, this may or may not be helpful: https://forums.linuxmint.com/viewtopic.php?t=250485

**Comment by vaniusrb** [comment]
Hi, guys! Has solved for me! Change shadow configuration and adding "nomodeset" has no effect. But I changed at login screen to "Cinnamon (Default)" instead "Cinnamon (Software Rendering)":

![login1](https://user-images.githubusercontent.com/11839284/58371834-8973e900-7eeb-11e9-9d2e-63df6569332f.jpeg)
![login2](https://user-images.githubusercontent.com/11839284/58371835-8b3dac80-7eeb-11e9-9b76-98fa83bd11bf.jpeg)

Probably I switched this option to solve an older problem.

**Comment by scoson** [comment]
@vaniusrb that's great if it was as simple as that...

@jaszhix I started performing a clean re-install before your last post (it needed it anyway imo). I have installed all updates to be current, configured my panel and all applets as desired, disabled effects, and am using custom theming elements. All is well after reboots. Your taking time to help is much appreciated and I have created a txt of your suggestions for later use if needed if it acts up again. So thank you jaszhix and also vaniusrb for giving me a nudge to post here..
